private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts

```diff
@@ -0,0 +1,170 @@
+import { ClientOptions } from "./IClientOptions";
+import express, { Request, Response, NextFunction, Router } from "express";
+import { check, validationResult } from "express-validator";
+import { AuthenticationProvider } from "."; 
+import { TemplateError, ApiError } from "./api/TemplateError";
+import { StorageProvider } from ".";
+import { ITemplate, JSONResponse, ITemplateInstance } from ".";
+
+export class TemplateServiceClient {
+
+    private static storageProvider: StorageProvider;
+    private static authProvider: AuthenticationProvider;
+
+    /**
+     * @public
+     * Initialize database if not already running
+     * @param {ClientOptions} clientOptions - storage provider and auth provider options
+     */
+    public static init(clientOptions: ClientOptions): TemplateServiceClient {
```

Comment:
are you planning to instantiate TemplateClient through this only? Consider hiding the constructor then?

Reply:
Fixed, thanks!